### PR TITLE
Include invocation reasons when analyzing documents by solution crawler

### PR DESCRIPTION
**Pull request by drognanar**

### CODE REVIEW

**Comment by drognanar** [comment]
FYI @KevinH-MS @mavasani @heejaechang

File: src/VisualStudio/Core/Def/Implementation/SolutionSize/SolutionSizeTracker.cs [hunk 1]
```diff
@@ -135,14 +136,19 @@ public bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs
...
+            public Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
             {
                 return SpecializedTasks.EmptyTask;
             }
 
             public void RemoveProject(ProjectId projectId)
             {
             }
+
+            internal Task AnalyzeSyntaxAsync(Document newDocument, object invocationReasons, CancellationToken none)
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/Implementation/SolutionSize/SolutionSizeTracker.cs]: Is this needed?

**Comment by heejaechang** [comment]
except what @mavasani pointed out, 👍

**Comment by KevinH-MS** [comment]
👍

**Pull request merged**